src/lightning_lite/strategies/fairscale.py

```diff
@@ -0,0 +1,125 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import operator
+from contextlib import contextmanager
+from datetime import timedelta
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
+
+import torch
+from lightning_utilities.core.imports import compare_version, module_available
+from torch.distributed.constants import default_pg_timeout
+from torch.nn import Module
+from torch.optim import Optimizer
+
+from lightning_lite.accelerators import Accelerator
+from lightning_lite.plugins import CheckpointIO, ClusterEnvironment, Precision
+from lightning_lite.strategies.ddp import DDPStrategy
+from lightning_lite.utilities.enums import PrecisionType
+from lightning_lite.utilities.imports import _IS_WINDOWS
+
+_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and module_available("fairscale.nn")
+_FAIRSCALE_OSS_FP16_BROADCAST_AVAILABLE = _FAIRSCALE_AVAILABLE and compare_version("fairscale", operator.ge, "0.3.3")
```

Comment:
Our min supported fairscale is higher than 0.3.3

```suggestion
```

I removed this in #14476 so must be a bad merge

Reply:
Okay, even better

Reply:
Ok, checked the PR. Nothing else was missed, only these imports in the strategy here directly. updated